mongodb/src/main/java/org/apache/zeppelin/mongodb/MongoDbInterpreter.java

```diff
@@ -0,0 +1,209 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * MongoDB interpreter. It uses the mongo shell to interpret the commands.
+ */
+public class MongoDbInterpreter extends Interpreter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongoDbInterpreter.class);
+
+  private static final String SHELL_EXTENSION =
+      new Scanner(MongoDbInterpreter.class.getResourceAsStream("/shell_extension.js"), "UTF-8")
+      .useDelimiter("\\A").next();
+
+  private static final int SIGTERM_CODE = 143;
+
+  private long commandTimeout = 60000;
+
+  private String dbAddress;
+
+  private int maxConcurrency = 10;
+
+  private Map<String, Executor> runningProcesses =  new HashMap<>();
+
+  public MongoDbInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    commandTimeout = Long.parseLong(getProperty("mongo.shell.command.timeout"));
+    maxConcurrency = Integer.parseInt(getProperty("mongo.interpreter.concurrency.max"));
+    dbAddress = getProperty("mongo.server.host") + ":"
+      + getProperty("mongo.server.port") + "/"
+      + getProperty("mongo.server.database");
+  }
+
+  @Override
+  public void close() {
+    runningProcesses.clear();
+    runningProcesses = null;
+  }
+
+  @Override
+  public FormType getFormType() {
+    return FormType.SIMPLE;
+  }
+
+  @Override
+  public InterpreterResult interpret(String script, InterpreterContext context) {
+    LOGGER.debug("Run MongoDB script: {}", script);
+
+    if (StringUtils.isEmpty(script)) {
+      return new InterpreterResult(Code.SUCCESS);
+    }
+
+    String paragraphId = context.getParagraphId();
+    // Write script in a temporary file
+    // The script is enriched with extensions
+    final File scriptFile = new File(getScriptFileName(paragraphId));
+    try {
+      FileUtils.write(scriptFile,
+          SHELL_EXTENSION
+          .replace("__ZEPPELIN_TABLE_LIMIT__", getProperty("mongo.shell.command.table.limit")) +
+          script);
+    }
+    catch (IOException e) {
+      LOGGER.error("Can not write script in temp file", e);
+      return new InterpreterResult(Code.ERROR, e.getMessage());
+    }
+
+    InterpreterResult result = new InterpreterResult(InterpreterResult.Code.SUCCESS);
+
+    final DefaultExecutor executor = new DefaultExecutor();
+    final ByteArrayOutputStream errorStream = new ByteArrayOutputStream();
+
+    executor.setStreamHandler(new PumpStreamHandler(context.out, errorStream));
+    executor.setWatchdog(new ExecuteWatchdog(commandTimeout));
+
+    final CommandLine cmdLine = CommandLine.parse(getProperty("mongo.shell.path"));
+    cmdLine.addArgument("--quiet", false);
+
+    if (!StringUtils.isEmpty(getProperty("mongo.server.username"))) {
+      cmdLine.addArgument("-u", false);
+      cmdLine.addArgument(getProperty("mongo.server.username"), false);
+      cmdLine.addArgument("-p", false);
+      cmdLine.addArgument(getProperty("mongo.server.password"), false);
```

Comment:
We should be really careful with passing passwords on the command-line, because they are visible in the `ps` output, etc. Maybe we can pass credentials as part of the script to execute? like this:

```
use admin
db.auth("uname", "password")
```

Reply:
I don't get your point. If we want the data to be secure, we need to use --tls option in the mongo shell. `CommandLine` with option --quiet will be less chatty, not visible in the `ps` output. Although this interpreter has not provided the secure TLS option, it's a simple one.

Reply:
my point is that people with shell access, for example, via shell interpreter, could do `ps -aef|grep mongo` and get the full command line with username & password.

Reply:
ok,I'll try